landlab/grid/create.py

```diff
@@ -84,3 +111,252 @@ def create_and_initialize_grid(input_source):
 
     # Return the created and initialized grid
     return grid_reader(param_dict)
+
+
+def create_grid(file_like):
+    """Create grid, initialize fields, and set boundary conditions.
+
+    **create_grid** expects a dictionary with three keys "grid", "fields", and
+    "boundary_conditions".
+
+    **Dictionary Section "grid"**
+
+    The value associated with the "grid" key should itself be a dictionary
+    containing the name of a Landlab model grid type as its only key. The
+    following grid types are valid:
+
+        -  :py:class:`~landlab.grid.raster.RasterModelGrid`
+        -  :py:class:`~landlab.grid.voronoi.VoronoiDelaunayGrid`
+        -  :py:class:`~landlab.grid.hex.HexModelGrid`
+        -  :py:class:`~landlab.grid.radial.RadialModelGrid`
+        -  :py:class:`~landlab.grid.network.NetworkModelGrid`
+
+    The value associated with the grid name key is a list containing the
+    arguments. If any keyword arguments are passed, they should be passed as
+    the last element of the list. For example the following code block is a
+    yaml file indicating a RasterModelGrid with shape (4, 5) and xy-spacing of
+    (3, 4).
+
+    .. code-block:: yaml
+
+        grid:
+          RasterModelGrid:
+            - [4, 5]
+            - xy_spacing: [3, 4]
+
+    These arguments and keyword arguments will be passed to the ``__init__``
+    constructor of the specified model grid. Refer to the documentation for
+    each grid to determine its requirements.
+
+    **Dictionary Section "fields"**
+
+    Fields can be created by reading from files or by creating synthetic
+    values.
+
+    The value associated with the "fields" key is a nested set of dictionaries
+    indicating where the fields are created, what the field names are, and how
+    to create the fields. At the highest hierachical level, the value
+    associated with the "fields" key must be a dictionary with keys indicating
+    at which grid elements fields should be created (e.g. to create fields at
+    node, use "at_node").
+
+    The value associated with each "at_xxx" value is itself a dictionary
+    indicating the name of the field an how it should be created. A field can
+    either be created by reading from a file or creating synthetic values. The
+    :py:func:`~landlab.io.netcdf.read.read_netcdf` and
+    :py:func:`~landlab.io.esri_ascii.read_esri_ascii` functions, and the
+    :py:mod:`synthetic fields <landlab.values.synthetic>`
+    package are currently supported methods to create fields. These may be
+    chained together (as is shown in the Example section below). If these
+    functions do not meet your needs, we welcome contributions that extend the
+    capabilities of this function.
+
+    The following example would uses the
```

Comment:
Add text explaining the usage of plane and normal

Reply:
Done! Good idea!